### ISSUE TITLE
Building for Android with ANDROID_PLATFORM < 24 results in fseeko() and ftello() always return error in 32-bit build

### ISSUE THREAD

**Issue by svenoaks** [xiph/flac#494]
The [32-bit ABI bugs document](https://android.googlesource.com/platform/bionic/+/master/docs/32-bit-abi.md) suggests they're not available, but I didn't notice it until encoder `finish()` is always false and metadata not written, it seems like `fseeko()` is always returning error.

I'm not sure how to fix it yet, I can't raise my ANDROID_PLATFORM to 24 yet.

**Comment by ktmf01** [comment]
A stupid question perhaps, but what would one gain by building libFLAC on Android? libFLAC is already part of the Android system API's.

Related to that: perhaps there is a patch somewhere for this in the Android sources? https://android.googlesource.com/platform/external/flac/

**Comment by svenoaks** [comment]
> A stupid question perhaps, but what would one gain by building libFLAC on Android? libFLAC is already part of the Android system API's.
> 
> Related to that: perhaps there is a patch somewhere for this in the Android sources? https://android.googlesource.com/platform/external/flac/

As far as I can tell, libFLAC can't be used directly by app developers, you have to go through through the Media API, which is cumbersome. I'll look more, but can't see how (if?) they resolved the fseeko thing there

**Comment by svenoaks** [comment]
I added this to compat.h 

```
#if defined __ANDROID__ && (defined __arm__ || defined __i386__) && __ANDROID_API__ < 24
#define fseeko fseek
#define ftello ftell
#endif
```

and it works, but I think that will fail with files larger than 2GB.

**Comment by ktmf01** [comment]
I can't seem to find any changes in Android's version of FLAC related to this.

Perhaps it is interesting to try and generate a FLAC file through some existing Android app using the Media API? If such a file also lacks the metadata you reference, perhaps this isn't fixed in the Media API either?

Anyway, I can't be of much help here, sadly.

**Comment by ktmf01** [comment]
I've been looking through the libFLAC that shipped with Android 6.0 (which shipped with API level 23) at https://android.googlesource.com/platform/external/flac/+/refs/tags/android-6.0.0_r26/

I can't find anything relating to this. It seems fseeko is used at several places. So, it seems fseeko wasn't made available but is available at a lower level?

Can I close this issue? You've found a fix (from the document you linked it seems the only possible fix).

**Comment by svenoaks** [comment]
> I've been looking through the libFLAC that shipped with Android 6.0 (which shipped with API level 23) at https://android.googlesource.com/platform/external/flac/+/refs/tags/android-6.0.0_r26/
> 
> I can't find anything relating to this. It seems fseeko is used at several places. So, it seems fseeko wasn't made available but is available at a lower level?
> 
> Can I close this issue? You've found a fix (from the document you linked it seems the only possible fix).

Fair enough, since devs compiling flac with ANDROID_PLATFORM < 24 and 32-bit Android in general is getting less and less common we shouldn't waste anymore brain power on it.
